wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceManager.java

```diff
@@ -243,12 +270,21 @@ public WorkspaceImpl updateWorkspace(String id, Workspace update)
       throws ConflictException, ServerException, NotFoundException, ValidationException {
     requireNonNull(id, "Required non-null workspace id");
     requireNonNull(update, "Required non-null workspace update");
-    requireNonNull(update.getConfig(), "Required non-null workspace configuration update");
-    validator.validateConfig(update.getConfig());
+    checkArgument(
+        update.getConfig() != null || update.getDevfile() != null,
+        "Required non-null workspace configuration or devfile update");
```

Comment:
Does it makes sense to check also that both config + devfile are not present?

Reply:
Makes sense. Rework to use `^` instead of `||`. Also added the same check to WorkspaceService.